mne/preprocessing/maxwell.py

```diff
@@ -171,16 +174,16 @@ def maxwell_filter(raw, origin=(0, 0, 40), int_order=8, ext_order=3,
 
         # Loop through buffer windows of data
         for win in zip(lims[:-1], lims[1:]):
-
             # Compute SSP-like projector. Set overlap limit to 0.02
-            proj = _overlap_projector(raw_sss._data[meg_chs, win[0]:win[1]],
-                                      resid[:, win[0]:win[1]], 0.02,
-                                      (win[0] / raw.info['sfreq'],
-                                       win[1] / raw.info['sfreq']))
-
+            this_data = raw_sss._data[meg_chs, win[0]:win[1]]
+            V = _overlap_projector(this_data, resid[:, win[0]:win[1]],
+                                   st_corr)
             # Apply projector according to Eq. 12 in [2]_
-            raw_sss._data[:, win[0]:win[1]] = \
-                np.dot(proj, raw_sss._data[:, win[0]:win[1]].T).T
+            logger.info('    Projecting out %s tSSS components for %s-%s'
+                        % (V.shape[1], win[0] / raw.info['sfreq'],
+                           win[1] / raw.info['sfreq']))
+            this_data -= np.dot(np.dot(this_data, V), V.T)
```

Comment:
This is the equivalent step. In your code before this was equivalent to np.dot(this_data, np.dot(V, V.T)), and that second term was huge (n_times x n_times). By changing the order of the dot products, we avoid that blowup, saving a ton of memory.